core/config.go

```diff
@@ -229,6 +229,11 @@ func (c *CLab) createNodeCfg( //nolint: funlen
 		Components:      c.Config.Topology.GetComponents(nodeName),
 	}
 
+	if nodeDef != nil {
```

Comment:
these parts (basically most of the changes in core/config.go) are lame. I feel like we have a bit of a silly pattern here where we have the config node then try to access the node def from the topo and it's just a bit wonky. I think the safety checks put in in this pr are fine for now but just reswizzling node instantiation to be safer in general in the future prolly would be good